sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSession;
+    };
+};
+
+// @public (undocumented)
+export type GetSparkStatementOptions = OperationOptions;
+
+// @public
+export type GetSparkStatementResponse = SparkStatement & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatement;
+    };
+};
+
+// @public (undocumented)
+export interface ListPageSettings {
+    continuationToken?: string;
+}
+
+// @public
+export interface ListSparkBatchJobsOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+    fromParam?: number;
+    size?: number;
+}
+
+// @public
+export type ListSparkBatchJobsResponse = SparkBatchJobCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkBatchJobCollection;
+    };
+};
+
+// @public
+export interface ListSparkSessionsOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+    fromParam?: number;
+    size?: number;
+}
+
+// @public
+export type ListSparkSessionsResponse = SparkSessionCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkSessionCollection;
+    };
+};
+
+// @public (undocumented)
+export type ListSparkStatementsOptions = OperationOptions;
+
+// @public
+export type ListSparkStatementsResponse = SparkStatementCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatementCollection;
+    };
+};
+
+// @public
+export const logger: import("@azure/logger").AzureLogger;
+
+// @public (undocumented)
+export interface OperationResponse {
+    _response: HttpResponse;
+}
+
+export { PipelineOptions }
+
+// @public
+export type PluginCurrentState = "Preparation" | "ResourceAcquisition" | "Queued" | "Submission" | "Monitoring" | "Cleanup" | "Ended" | string;
+
+// @public (undocumented)
+export type ResetSparkSessionTimeoutOptions = OperationOptions;
+
+// @public
+export type SchedulerCurrentState = "Queued" | "Scheduled" | "Ended" | string;
+
+// @public (undocumented)
+export interface SparkBatchJob {
+    appId?: string;
+    appInfo?: {
+        [propertyName: string]: string;
+    };
+    artifactId?: string;
+    errors?: SparkServiceError[];
+    id: number;
+    jobType?: SparkJobType;
+    // (undocumented)
+    livyInfo?: SparkBatchJobState;
+    logLines?: string[];
+    name?: string;
+    plugin?: SparkServicePlugin;
+    result?: SparkBatchJobResultType;
+    scheduler?: SparkScheduler;
+    sparkPoolName?: string;
+    state?: string;
+    submitterId?: string;
+    submitterName?: string;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    workspaceName?: string;
+}
+
+// @public
+export interface SparkBatchJobCollection {
+    from: number;
+    sessions?: SparkBatchJob[];
+    total: number;
+}
+
+// @public (undocumented)
+export interface SparkBatchJobOptions {
```

Comment:
a lot of overlap with `SparkBatchJob` - can we extend one into the other?

Reply:
I will try to do that at convenient layer.

---
In reply to: [521611031](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521611031) [](ancestors = 521611031)